### PR TITLE
fix the issue that we cannot get namespace of `Azure.Core` and `Azure.ResourceManager` types in contract

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Prerequisite of fixing https://github.com/Azure/autorest.csharp/issues/3361

# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Common/AutoRest/Plugins/GeneratedCodeWorkspace.cs [hunk 1]
```diff
@@ -24,6 +25,25 @@ internal class GeneratedCodeWorkspace
...
         public static readonly string GeneratedFolder = "Generated";
 
+        private static readonly IReadOnlyList<MetadataReference> AssemblyMetadataReferences;
+
+        static GeneratedCodeWorkspace()
+        {
+            var references = new List<MetadataReference>
+            {
+                MetadataReference.CreateFromFile(typeof(object).Assembly.Location),
+                MetadataReference.CreateFromFile(typeof(Response).Assembly.Location),
```
Comment (pshao25) [inline src/AutoRest.CSharp/Common/AutoRest/Plugins/GeneratedCodeWorkspace.cs:35]: So you choose an arbitrary type from `azure.core`, in this case it it `Response`?

Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Common/AutoRest/Plugins/GeneratedCodeWorkspace.cs:35]: Yes, to dynamically get the location of the assembly of the dll

**Review by pshao25 — APPROVED** [review]

**Comment by lirenhe** [comment]
@ArcturusZhang, is there any issue for this PR and could you add test case for it?

**Comment by ArcturusZhang** [comment]
> @ArcturusZhang, is there any issue for this PR and could you add test case for it?

It involves a lot of context things therefore it is quite difficult to introduce a test case for this.
But I could create an issue for the issue this PR is solving

**Comment by ArcturusZhang** [comment]
> @ArcturusZhang, is there any issue for this PR and could you add test case for it?

Issue is here: https://github.com/Azure/autorest.csharp/issues/3361
This PR is not fixing it, this PR is the prerequisite of fixing it.

**Review by lirenhe — APPROVED** [review]

**Pull request merged**